### PR TITLE
Azure DevOps fixes

### PR DESCRIPTION
**Pull request by niall-turbitt**
### Summary of Azure DevOps fixes/updates:

- Fix triggers for CI/CD pipelines. Previously, pipelines were triggered on pushes to any branch (`IndividualCI`). Now CI pipelines are only triggered on PR to default branch. Terraform deploy now triggers when PR has been successfully merged to default branch (and only on changes to `/databricks/config/**`) [[ML-27221]](https://databricks.atlassian.net/browse/ML-27221)
  - Added required permissions for build service to enable this. We set up build validation on the default branch in terraform cicd boostrap. In order for the Terraform deploy step to be able to write to the default branch after successfully merging a PR to main, we must give the build service permissions to override this branch policy. [[ML-27222]](https://databricks.atlassian.net/browse/ML-27222)
- Version requirements for terraform, databricks-cli, azure-cli [[ML-27220]](https://databricks.atlassian.net/browse/ML-27220)
- List explicit PAT permissions required [[ML-27219]](https://databricks.atlassian.net/browse/ML-27219)
- Change Terraform output variables to camelCase in order to make them compatible with Azure Key Vault [[ML-27218]](https://databricks.atlassian.net/browse/ML-27218)

### CODE REVIEW

**Review by vladimirk-db — APPROVED** [review]

**Comment by niall-turbitt** [comment]
### Summary of Azure DevOps fixes/updates:

- Fix triggers for CI/CD pipelines. Previously, pipelines were triggered on pushes to any branch (`IndividualCI`). Now CI pipelines are only triggered on PR to default branch. Terraform deploy now triggers when PR has been successfully merged to default branch (and only on changes to `/databricks/config/**`) [[ML-27221]](https://databricks.atlassian.net/browse/ML-27221)
  - Added required permissions for build service to enable this. We set up build validation on the default branch in terraform cicd boostrap. In order for the Terraform deploy step to be able to write to the default branch after successfully merging a PR to main, we must give the build service permissions to override this branch policy.
- Version requirements for terraform, databricks-cli, azure-cli [[ML-27220]](https://databricks.atlassian.net/browse/ML-27220)
- List explicit PAT permissions required [[ML-27219]](https://databricks.atlassian.net/browse/ML-27219)
- Change Terraform output variables to camelCase in order to make them compatible with Azure Key Vault [[ML-27218]](https://databricks.atlassian.net/browse/ML-27218)

File: {{cookiecutter.project_name}}/.mlops-setup-scripts/README.md
Comment (vladimirk-db) [inline {{cookiecutter.project_name}}/.mlops-setup-scripts/README.md:160]: nit for consistency and clarity: please use "you" as opposed to "we" for actions that the user is supposed to perform.

File: {{cookiecutter.project_name}}/.azure/devops-pipelines/terraform-cicd.yml
Comment (niall-turbitt) [reply {{cookiecutter.project_name}}/.azure/devops-pipelines/terraform-cicd.yml]: yep bug, thanks for catching. Have just pushed the change and test out setting up

File: {{cookiecutter.project_name}}/.azure/devops-pipelines/tests-ci.yml
Comment (niall-turbitt) [reply {{cookiecutter.project_name}}/.azure/devops-pipelines/tests-ci.yml:17]: done!

File: {{cookiecutter.project_name}}/.mlops-setup-scripts/README.md
Comment (niall-turbitt) [reply {{cookiecutter.project_name}}/.mlops-setup-scripts/README.md]: done!

**Review by jerrylian-db — COMMENTED** [review]
LGTM! I'll approve the PR after I do a CUJ with the branch code.

**Comment by niall-turbitt** [comment]
variable ARM_SUBSCRIPTION_ID is required in the terraform_cicd yml due to changing the variable name in the variable group from ARM_SUBSCRIPTION_ID to armSubscriptionId (required to be compatible with Key Vault)

See [this issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/3772) for more details

**Comment by niall-turbitt** [comment]
@vladimirk-db tested setting up from scratch with `cookiecutter https://github.com/niall-turbitt/mlops-stack.git -c niall-turbitt/ado-fixes` and working for CUJs for changing ML code, and changing `databricks-config/**` code

**Review by vladimirk-db — APPROVED** [review]
Thanks @niall-turbitt !

Please wait for @jerrylian-db's LGTM as well.

File: {{cookiecutter.project_name}}/.mlops-setup-scripts/README.md
Comment (jerrylian-db) [inline {{cookiecutter.project_name}}/.mlops-setup-scripts/README.md:229]: Here is how this is rendered. Can you fix?
Screen Shot 2022-12-06 at 6 03 44 PM

**Review by jerrylian-db — APPROVED** [review]
LGTM! Verified by dogfooding the PR branch myself, that the PR does what it intended to do. Great work! Feel free to merge after addressing the doc nit.

**Comment by jerrylian-db** [comment]
@niall-turbitt Made some small fixes and merging this PR. Thanks for your work!

**Pull request merged**
